python/GafferUI/ColorChooser.py

```diff
@@ -389,49 +405,173 @@ def __positionToColor( self, position ) :
 		if self.__useWheel() :
 			halfSize = imath.V2f( size ) * 0.5
 			p = self.__cartesianToPolar( imath.V2f( position.x, size.y - position.y ), halfSize, halfSize.x )
-			c[xIndex] = p.x
+			if p.y > 0 :
+				c[xIndex] = p.x
 			c[yIndex] = p.y
 		else :
 			c[xIndex] = ( position.x / float( size.x ) ) * ( _ranges[xComponent].max - _ranges[xComponent].min ) + _ranges[xComponent].min
 			c[yIndex] = ( 1.0 - ( position.y / float( size.y ) ) ) * ( _ranges[yComponent].max - _ranges[yComponent].min ) + _ranges[yComponent].min
 
 		return c
 
+	def __clampPosition( self, position ) :
+
+		size = self.bound().size()
+		if self.__useWheel() :
+			halfSize = imath.V2f( size ) * 0.5
+			centeredPosition = position - halfSize
+			radiusOriginal = centeredPosition.length()
+
+			if radiusOriginal == 0 :
+				return halfSize
+
+			return halfSize + ( centeredPosition / imath.V2f( radiusOriginal )  ) * min( radiusOriginal, halfSize.x )
+
+		return imath.V2f( min( size.x, max( 0.0, position.x ) ), min( size.y, max( 0.0, position.y ) ) )
+
+	def __constrainPosition( self, position ) :
+
+		if self.__dragConstraints is None :
+			return position
+
+		assert( bool( self.__dragConstraints ) )
+
+		if not self.__useWheel() :
+			# Determine if `position` is closer to the horizontal line going through
+			# `__constraintPosition` or the vertical line. The distance to the
+			# horizontal line is the vertical difference from `position` to
+			# `__constraintPosition`. The distance to the vertical line is the
+			# horizontal difference.
+			delta = self.__constraintPosition - position
+			delta.x = abs( delta.x )
+			delta.y = abs( delta.y )
+
+			if delta.y < delta.x and (
+				self.__DragConstraints.X in self.__dragConstraints or not self.__DragConstraints.Y in self.__dragConstraints
+			) :
+				# Constrain to horizontal line
+				return imath.V2f( position.x, self.__constraintPosition.y )
+			# Constrain to vertical line
+			return imath.V2f( self.__constraintPosition.x, position.y )
+
+		center = imath.V2f( self.bound().size() ) * 0.5
+		distanceToCenter = ( position - center ).length()
+
+		if distanceToCenter < self.__snapRadius :
+			# If we're very close the center, snap to the center.
+			return center
```

Comment:
Do you find this center snap to be a useful interaction? I wonder if it's necessary given the latest round of changes making it easier to reach 0 by dragging past the origin of the line without immediately snapping to the circle.

With the increased `__snapRadius` in the latest fixups it means that I struggle to set a constrained value or saturation below 0.133 on a 7 slider tall color wheel as this takes over and snaps me straight to 0.